devicetypes/smartthings/ozom-smart-siren.src/ozom-smart-siren.groovy

```diff
@@ -127,32 +146,49 @@ def configure() {
 
 def both() {
 	log.debug "both()"
-	on()
+	state.alarmCmd = ALARM_BOTH
+	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
+	state.lastDuration = warningDuration
+
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, MODE_BOTH, DataType.pack(warningDuration, DataType.UINT16), STROBE_DUTY_CYCLE, STROBE_LEVEL)
 }
 
 def siren() {
 	log.debug "siren()"
-	on()
+
+	state.alarmCmd = ALARM_SIREN
+	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
+	state.lastDuration = warningDuration
+
+	// start warning, burglar mode, no strobe, siren very high
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, MODE_SIREN, DataType.pack(warningDuration, DataType.UINT16), "00", "00")
 }
 
 def strobe() {
-	log.warn "strobe() is not supported by this device"
+	log.debug "strobe()"
+
+	state.alarmCmd = ALARM_STROBE
+	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
+	state.lastDuration = warningDuration
+
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, MODE_STROBE, DataType.pack(warningDuration, DataType.UINT16), STROBE_DUTY_CYCLE, STROBE_LEVEL)
 }
 
 def on() {
 	log.debug "on()"
 
-	state.alarmOn = true
-	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
-	state.lastDuration = warningDuration
+	def manufacturer = device.getDataValue("manufacturer")
 
-	// start warning, burglar mode, no strobe, siren very high
-	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, "13", DataType.pack(warningDuration, DataType.UINT16), "00", "00")
+	if (manufacturer == "ClimaxTechnology"){
```

Comment:
Generally we've moved this sort of logic in to helper functions so that it is easier to re-use. There's only one use here, so it might be fine. But this is usually what it looks like:

```
private isOzomSiren() {
device.getDataValue("manufacturer") == "ClimaxTechnology"
}

...

if (isOzomSiren()) {
// do stuff
}
```

Like here: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zigbee-battery-accessory-dimmer.src/zigbee-battery-accessory-dimmer.groovy#L75

Reply:
just modified follow your guide